react-accordion/my-app/src/Accordion.tsx

```diff
@@ -0,0 +1,33 @@
+import { TopicCard } from './TopicCard';
+import { useState } from 'react';
+
+type Topics = {
+  id: number;
+  title: string;
+  content: string;
+};
+
+type Props = {
+  topics: Topics[];
+};
+
+export function Accordion({ topics }: Props) {
+  const [openTopicId, setOpenTopicId] = useState<number | null>(null);
```

Comment:
I find it more convenient to use `undefined` instead of `null`.

I actually heard Brendan Eich, the creator of JavaScript, once admit that `null` was a mistake and should never have been introduced into the language. So I use it only when I have to (usually because someone else used it, like the DOM).

On a similar note, you could just use `id === 0` to indicate "no topic selected". In pretty much every data management system in existence, 0 is considered to be an illegal ID, so you can use it to mean "none".